botocore/utils.py

```diff
@@ -645,7 +646,8 @@ def fix_s3_host(request, signature_version, region_name, **kwargs):
     """
     # By default we do not use virtual hosted style addressing when
     # signed with signature version 4.
-    if signature_version in ['s3v4', 'v4']:
+    if signature_version is not botocore.UNSIGNED and \
+            's3v4' in signature_version:
```

Comment:
Ah the handlers protect it: https://github.com/boto/botocore/blob/develop/botocore/handlers.py#L820. Yeah that's fine then.